plugins/inputs/gnmi/handler.go

```diff
@@ -98,21 +111,50 @@ func (h *handler) subscribeGNMI(ctx context.Context, acc telegraf.Accumulator, t
 				h.log.Debugf("update_%v: %s", t, string(buf))
 			}
 		}
-
 		if response, ok := reply.Response.(*gnmiLib.SubscribeResponse_Update); ok {
-			h.handleSubscribeResponseUpdate(acc, response)
+			h.handleSubscribeResponseUpdate(acc, response, reply.GetExtension())
 		}
 	}
 	return nil
 }
 
 // Handle SubscribeResponse_Update message from gNMI and parse contained telemetry data
-func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, response *gnmiLib.SubscribeResponse_Update) {
+func (h *handler) handleSubscribeResponseUpdate(acc telegraf.Accumulator, response *gnmiLib.SubscribeResponse_Update, extension []*gnmiExt.Extension) {
 	var prefix, prefixAliasPath string
 	grouper := metric.NewSeriesGrouper()
 	timestamp := time.Unix(0, response.Update.Timestamp)
 	prefixTags := make(map[string]string)
 
+	// iter on each extension
+	for _, ext := range extension {
+		currentExt := ext.GetRegisteredExt().Msg
+		if currentExt == nil {
+			break
+		}
+		// extension ID
+		switch ext.GetRegisteredExt().Id {
+		// Juniper Telemetry header
+		//EID_JUNIPER_TELEMETRY_HEADER = 1;
+		case 1:
+			// Decode it only if user requested it
+			if h.jnprExtension {
+				juniperHeader := &jnprHeader.GnmiJuniperTelemetryHeader{}
+				// unmarshal extention
+				err := proto.Unmarshal(currentExt, juniperHeader)
+				if err != nil {
+					break
+				}
+				// Add only relevant Tags from the juniper extension header.
+				// These are requiered for aggregation
+				prefixTags["component_id"] = fmt.Sprint(juniperHeader.GetComponentId())
+				prefixTags["component"] = fmt.Sprint(juniperHeader.GetComponent())
+				prefixTags["sub_component_id"] = fmt.Sprint(juniperHeader.GetSubComponentId())
```

Comment:
Talk to me about these IDs. Are these like an index that can grow forever? If so, these could introduce some cardinality issues. If it is the sort of thing that is bounded or normally less than a certain number then this would be ok.

Reply:
This id are quite limited. In fact, the component_id max is 20 and sub_component_id max 12. I've never experienced any cardinality issues at scales with hundreds of routers